tests/integration/test_broken_detached_part_clean_up/test.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+
+from helpers.cluster import ClickHouseCluster
+from multiprocessing.dummy import Pool
+from helpers.corrupt_part_data_on_disk import corrupt_part_data_on_disk
+from helpers.corrupt_part_data_on_disk import break_part
+import time
+
+cluster = ClickHouseCluster(__file__)
+
+node1 = cluster.add_instance("node1", stay_alive=True, with_zookeeper=True)
+
+path_to_data = "/var/lib/clickhouse/"
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+def remove_broken_detached_part_impl(table, node, expect_broken_prefix):
+    assert (
+        node.query(
+            f"SELECT COUNT() FROM system.parts WHERE table='{table}' AND active=1"
+        )
+        == "4\n"
+    )
+
+    path_to_detached = path_to_data + f"data/default/{table}/detached/"
+
+    result = node.exec_in_container(["ls", path_to_detached])
+    assert result.strip() == ""
+
+    corrupt_part_data_on_disk(node, table, "all_3_3_0")
+    break_part(node, table, "all_3_3_0")
+    node.query(f"ALTER TABLE {table} DETACH PART 'all_1_1_0'")
+    result = node.exec_in_container(["touch", f"{path_to_detached}trash"])
```

Comment:
A test that moves a directory that was modified a long time ago is still required. Directory name should start with some "broken prefix".